### PR TITLE
Add ability to activate and deactivate matches

### PR DESCRIPTION
**Pull request by juliaramstedt**
- [x] When matches are created their "concluded_at" date is set to 6 months in the future
- [x] Adds a function for deactivating matchings, which sets a date on the column "concluded_at"
- [x] Adds a button in admin for deactivating matchings
- [x] Adds scopes to matchings to separate between active and inactive
- [x] Moves the "destroy" button so that only inactive matchings can be destroyed

### CODE REVIEW

File: admin/app/controllers/admin/matches_controller.rb [hunk 1]
```diff
@@ -4,6 +4,16 @@ module Admin
...
     include Godmin::Resources::ResourceController
 
+    def conclude
+      Match.find(params[:id]).conclude
+      redirect_to resource, notice: "Matchingen avslutades"
+    end
+
+    def reactivate
+      Match.find(params[:id]).reactivate
+      redirect_to resource, notice: "Matchingens avslut ångrades"
```
Comment (juliaramstedt) [inline admin/app/controllers/admin/matches_controller.rb]: maybe we should be using the redirect_after_update path. And put all language in one of the ymls...?

File: admin/config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,9 @@
 Admin::Engine.routes.draw do
   root to: "people#index"
-  resources :matches
+  resources :matches do
+    get "conclude", on: :member
+    get "reactivate", on: :member
```
Comment (juliaramstedt) [inline admin/config/routes.rb]: Naming feedback welcome!

File: admin/app/controllers/admin/matches_controller.rb (hunk 1 above)
Comment (jensljungblad) [inline admin/app/controllers/admin/matches_controller.rb]: Maybe do these as `create` and `destroy` on a `MatchConclusionController`? I know, I know, I'm obsessed with the REST-stuff...

**Comment by jensljungblad** [comment]
In light of what we talked about yesterday, should `concluded_at` be `concludes_at` and be present already from the beginning of a match, but set to 6 months in the future? Then when concluding a match I guess we could set the date to todays date like it is currently, and reactivate would calculate a new `concludes_at` date based on the `created_at` column.

Not sure, what do you think?

File: admin/app/controllers/admin/match_conclusions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require_dependency "admin/application_controller"
+
+module Admin
+  class MatchConclusionsController < ApplicationController
```
Comment (jensljungblad) [inline admin/app/controllers/admin/match_conclusions_controller.rb]: Such nice

**Comment by jensljungblad** [comment]
Should we add something to `Person#last_match` so that it checks for active matches?

File: admin/app/controllers/admin/matches_controller.rb [hunk 4]
```diff
@@ -4,10 +4,15 @@ module Admin
   class MatchesController < ApplicationController
     include Godmin::Resources::ResourceController
 
+    def create
+      @resource.concluded_at = 6.months.from_now
```
Comment (jensljungblad) [inline admin/app/controllers/admin/matches_controller.rb]: Kan använda `create_resource` i service objektet här

File: admin/config/routes.rb [hunk 5]
```diff
@@ -1,6 +1,8 @@
 Admin::Engine.routes.draw do
   root to: "people#index"
-  resources :matches
+  resources :matches do
+    resource :match_conclusions, only: [:create, :destroy]
```
Comment (jensljungblad) [inline admin/config/routes.rb]: No destroy

File: app/models/match.rb [hunk 6]
```diff
@@ -12,6 +12,25 @@ def matched_with(person)
     newcomer == person ? established : newcomer
   end
 
+  scope :active, -> { where("concluded_at >= ?", Time.zone.now) }
+  scope :inactive, -> { where("concluded_at < ?", Time.zone.now) }
+
+  def active?
+    concluded_at > Time.zone.now
```
Comment (jensljungblad) [inline app/models/match.rb]: Or equal to?

File: app/models/match.rb [hunk 7]
```diff
@@ -12,6 +12,25 @@ def matched_with(person)
...
+  end
+
+  def conclude!
```
Comment (jensljungblad) [inline app/models/match.rb]: No bang!

File: test/test_helper.rb [hunk 8]
```diff
@@ -3,8 +3,30 @@
 require 'rails/test_help'
```
Comment (jensljungblad) [inline test/test_helper.rb]: Delete

File: test/test_helper.rb [hunk 9]
```diff
@@ -3,8 +3,30 @@
...
+      age: 27,
+      email: "user_#{rand(5..30)}@example.com",
+      status: :newcomer,
+      engaged: false
+    )
+  end
+
+  def create_match
+    established = create_established_person
+    newcomer = create_established_person
```
Comment (jensljungblad) [inline test/test_helper.rb]: Newcomer!

File: test/test_helper.rb [hunk 10]
```diff
@@ -3,8 +3,30 @@
...
+    create_person(:newcomer)
+  end
+
+  def create_person(status)
+    Person.create!(
+      name: "Name",
+      gender: :female,
+      age: 27,
+      email: "user_#{rand(5..30)}@example.com",
+      status: :newcomer,
```
Comment (jensljungblad) [inline test/test_helper.rb]: Status!

**Pull request merged**